I18n Polishing for 4Minitz v2.0.0

derwok opened the pull request:
See issue #30 checklist:
https://github.com/4minitz/4minitz/issues/30#issuecomment-586582526

4minitz-bot committed:
Bot's first contribution

derwok committed:
I18N Test: Manual UI test checklist [skip ci]
All important UI visible texts as screen shots with needed preconditions
how to get there. For manual checking of new languages and changes.

derwok committed:
Remove i18n-bot test file

derwok committed:
I18N Test: additional screenshots [skip ci]

derwok committed:
i18n: More strings to resources.

derwok committed:
Travis-CI workaround
Error: Failed to fetch https://downloads.apache.org/cassandra/debian/dists/39x/main/binary-amd64/Packages  Protocol http not supported or disabled in libcurl

derwok committed:
i18n: More strings to resources

itrich reviewed (approved):
Looks fine to me! Thank you for further improvements to the new internalization feature! 👍

derwok commented:
> Looks fine to me! Thank you for further improvements to the new internalization feature! 👍

Great! Thanx for the feedback. End of the week I'll continue... getting strings as "final as possible" (TM). So we first bring German in shape and perform some testing. Then REAL translation may start on that base.

derwok committed:
Travis: Revert "Travis-CI workaround" - see if travis works again?

This reverts commit d5a0e04b8a5195daaee3d43b2bdab57c167e73ee.

itrich committed:
Added screenshots of mail for manual i18n testing (#531)

* Added screenshots of mail for manual i18n testing

* Fixed typo in i18n_manual-tests.md

derwok committed:
i18n: More strings to resources (by FLashMessage)

derwok committed:
i18n: Some minor fixes

derwok committed:
i18n: Removed triple curlies {{{...}}} that needed HTML formatting AND variables
As this could open #security attack vectors to render HTML with embedded user variables
Triple curly WITHOUT user variables still OK.
Solution is to embed user vars that need formatting in #markdown blocks.
As we shift markdown formatting to resource files, some simple
single-line 'confirm*' dialog templates not needed anymore.

derwok committed:
i18n: Plaintext eMails to resources

derwok committed:
i18n: Fix i18n missing YAML key error

derwok committed:
i18n: Pull finalize quality test results to resources.
Plus some other minor fixes, reported as warnings by node tests/i18n/test_i18n_resources.js
  I18N-Warning: >Dialog.ConfirmDeleteMinutes.title< from YAML never used in code.
  I18N-Warning: >Item.editInfo< from YAML never used in code.
  I18N-Warning: >Topic.delete< from YAML never used in code.
  I18N-Warning: >Minutes.unfinalizeHint< from YAML never used in code.
  I18N-Warning: >Minutes.finalizeHint< from YAML never used in code.
  I18N-Warning: >MeetingSeries.Edit.Error.notRegistered< from YAML never used in code.
  I18N-Warning: >MeetingSeries.Edit.Error.alreadyInList< from YAML never used in code.
  I18N-Warning: >Mail.newMinutes< from YAML never used in code.
  I18N-Warning: >Mail.AdminNewVersion.body< from YAML never used in code.
  I18N-Warning: >Mail.UserRoleChange.body2test< from YAML never used in code.

derwok committed:
i18n: Handle multiple prioritized browser languages
If browser has array of multiple preferred languages, we walk
down the list until we find the first locale code we support.
Fall back will stay with 'en-US'.

derwok committed:
i18n: More screen shots for manual test [skip ci]

derwok committed:
i18n: Modern browsers don't support custom messages for beforeunload event
So we remove it from code. So, we don't have to translate this string.
Details see: https://stackoverflow.com/a/37782307

derwok committed:
i18n: Fix codeclimate: reduce cyclomatic complexity
